instrumentation/wai/src/OpenTelemetry/Instrumentation/Wai.hs

```diff
@@ -52,48 +61,114 @@ newOpenTelemetryWaiMiddleware' tp =
       let path_ = T.decodeUtf8 $ rawPathInfo req
       -- peer = remoteHost req
       parentContextM
-      inSpan'' tracer path_ (defaultSpanArguments {kind = Server, attributes = usefulCallsite}) $ \requestSpan -> do
+      semConvStabilityOptIn <- getSemConvStabilityOptIn
+      let args =
+            defaultSpanArguments
+              { kind = Server
+              , attributes =
+                  case semConvStabilityOptIn of
+                    Stable ->
+                      usefulCallsite
+                        `H.union` [
+                                    ( "user_agent.original"
```

Comment:
This attribute must be provided to the sampler and is therefore has to be added when the span is created and not after.